### PR TITLE
Instructions and demo files for deploying Scout on container pods (Kubernetes use case)

### LINKED ISSUE #2186: Document how to create prod server and cli using containers -- Pods
Might need some changes in how the mongodb host is set --> It's not localhost anymore but the name of the mongodb service (if mongo is a pod service)

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #2186 

This PR offers a tutorial on how deploy Scout services as pods. Scout image is pulled from Docker Hub and used within Minikube to create containers for scout-web (server) and scout-cli(cli). Another container with an authenticated mongo instance is provided.

I chose to start working with Kubernetes pods because the software is much more documented than Podman.
Note that Kubernetes pods can be runned using Podman with the command `play kube`.

**How to test**:
1. Follow the instructions given in file: `docs/admin-guide/container-deploy.md` and using the files provided in the "container" folder.

**Expected outcome**:
1. You should be able to create pods and services to
- Run an authenticated mongo instance of a pod
- Run an interactive Scout cli on a pod
- Run the Scout web deployment on the pod
- Scout database can be populated with demo data (or any data)
- Once populated the database with demo data, the scout web app should be running normally from a browser window (with demo user and demo case etc).

Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR, DN

### CODE REVIEW

**Comment by northwestwitch** [comment]
Test creating and running the containers on Minikube: 🆗 

image

Test starting an interactive Scout cli shell (`kubectl exec -it scout-cli -- bash`) to populate db with demo data 🆗
image

Test populating the demo db with demo data: 🆗 
image

Test launching the web service (`minikube service cout-web-service`): 🆗 
image

image

**Comment by northwestwitch** [comment]
This is ready for review

**Comment by northwestwitch** [comment]
@eriksjolund I know that you are busy with many other things but if you have time I could use a review for this PR. Perhaps there is something we could use in our new VM!

**Review by dnil — APPROVED** [review]
🌟and🏅: this is really slick. Well done!

**Comment by dnil** [comment]
And, yeah, tested as well:
Screenshot 2021-01-14 at 14 02 05
Screenshot 2021-01-14 at 14 02 36

File: docs/admin-guide/container-deploy.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+### Overview of the running containers and starting the minikube service
+1. After the steps above a number of container should be running on the virtual machine. An overview of all containers together with accessory services and itams will be available using the command:
+```
+kubectl get all
+```
+
+1. Start the scout web app service using the command:
+```
+Minikube service scout-web-service
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply docs/admin-guide/container-deploy.md:180]: yes it is, thanks!

File: docs/admin-guide/container-deploy.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+type: Opaque
+data:
+  scout-username: c2NvdXRVc2Vy # base-64 encoded "scoutUser"
+  scout-password: c2NvdXRQYXNzd29yZA== # base-64 encoded "scoutPassword"
+```
+This secret should be edited to include the real scout-username and scout-password. Note that these fields should be base-64 encoded. To encode these strings you can run the following command:
+`echo -n "some string" | base64`
+
+1. **Create the above secrets** with the command:
+`kubectl apply -f secrets.yaml`
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply docs/admin-guide/container-deploy.md:58]: Well this depends on which directory you are when you run the command, what about path/to/secrets.yaml ?

[the remark this answers is not in the document]
Comment (dnil) [reply docs/admin-guide/container-deploy.md:58]: All, good, or ~/mysandbox/scout/scout/containers/kubernetes/...  - kinda just wanted to highlight that there are files included with the bistro if they tl;dr-ed it and started pasting to cat..

[the remark this answers is not in the document]
Comment (dnil) [reply docs/admin-guide/container-deploy.md:58]: *distro* 🤣

File: docs/admin-guide/container-deploy.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+### Requirements
+1. Install hyperkit and minikube on your local machine (OS X 10.10.3 Yosemite or later):
+```
+brew install hyperkit
+brew install minikube
+```
+The second command, brew install minikube, will install also the Kubernetes cli (kubectl). Minikube can be alternatively installed using another container manager, such as Podman or VMWare. More documentation is available [here](https://minikube.sigs.k8s.io/docs/start/).
+
+### Installation and Setup of the database and config files
+1. Start minikube kubernetes cluster inside hypervisor (hyperkit):
```
[the remark this answers is not in the document]
Comment (northwestwitch) [reply docs/admin-guide/container-deploy.md:24]: I haven't used Docker Desktop so I'm a bit confused on what to write about it. Could you help formulating it?

File: docs/admin-guide/container-deploy.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (northwestwitch) [reply docs/admin-guide/container-deploy.md:58]: I guess it was lunch time.. So what do you want to write?

[the remark this answers is not in the document]
Comment (dnil) [reply docs/admin-guide/container-deploy.md:58]: How about path/to/scout/containers/kubernetes?

**Review by dnil — APPROVED** [review]

File: docs/admin-guide/container-deploy.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (northwestwitch) [reply docs/admin-guide/container-deploy.md:58]: Yes!

File: docs/admin-guide/container-deploy.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+A local image of the repository can be created by moving the Dockerfile in the root folder of the app and from the same location, in a terminal, running the following command:
+
+docker build -t scout .
+
+The container with the docker image contains only the app installation files and its required libraries. In order to work, the container must be connected with at least one other container hosting a running mongodb instance or a database running on the local machine.
+
+
+## Example deploying Scout using Kubernetes
+Kubernetes is an open-source platform for managing containerized applications. This tutorial will show how to setup a virtual box containing a master and a work node on a local virtual machine using [minikube](https://github.com/kubernetes/minikube) and how to setup the necessary containers to run a MongoDB instance, a pod with the Scout command line and a pod with the Scout server.
```
Comment (dnil) [inline docs/admin-guide/container-deploy.md:13]:
```suggestion
Kubernetes is an open-source platform for managing containerized applications. This tutorial will show how to setup a virtual box containing a master and a work node on a local virtual machine using [minikube](https://github.com/kubernetes/minikube) and how to setup the necessary containers to run a MongoDB instance, a pod with the Scout command line and a pod with the Scout server on macOS. Alternatively, for macOS and Windows, consider  [Docker Desktop](https://www.docker.com/products/docker-desktop) which has builtin Kubernetes support. The default install locations for Docker Desktop do overlap with Homebrew hyperkit and minikube so it is easiest to pick one of the alternatives.
```

Comment (northwestwitch) [reply docs/admin-guide/container-deploy.md]: Thanks!

Comment (dnil) [reply docs/admin-guide/container-deploy.md]: Most welcome - I realised just now you had asked for it.. 😊

**Pull request merged**